docs/deployment_userinterface.md

```diff
@@ -0,0 +1,302 @@
+
+## System Requirements
+- The BrainKB UI is based on NextJS; we need to install the Node.js. When writing this document, the minimum version is **Node.js 18.17 or later**. For more (or latest) information, please check NextJS website.
+
+	- NextJS: [https://nextjs.org/docs/getting-started/installation](https://nextjs.org/docs/getting-started/installation)
+	- **<span style="color: red;">Note:</span>** Manual installation of latest version of NextJS.
+	
+		```
+		npm install next@latest react@latest react-dom@latest
+		```
+		If you get issue with depenency, add `--legacy-peer-deps` options.
+	- Installation of the dependencies.
+		```
+		npm install 
+		```
+
+- Once all the NextJS requirements has been met or covered, next step is to configure the OAuth. In our case we use Google and GitHub. Therefore, you would need to obtain the OAuth client ID and secret and update the ```.env.local``` file (shown below). Additionally, you also need to provide the secret for [NextAuth](https://next-auth.js.org/), a library that we use authentication. The ```NEXTAUTH_URL``` for development is ```http://localhost:3000```. Similarly, the GitHub secret and client ID can be obtained from GitHub at [https://github.com/settings/tokens](https://github.com/settings/tokens) and Google at [https://console.cloud.google.com/](https://console.cloud.google.com/).
+
+	```
+	#OAuth credentials
+	GITHUB_CLIENT_ID= 
+	GITHUB_CLIENT_SECRET=
+
+	GOOGLE_CLIENT_ID=
+	GOOGLE_CLIENT_SECRET=
+
+	#Auth Config
+	NEXTAUTH_SECRET= 
+	NEXTAUTH_URL=
+
+	# Query service API configuration 
+	NEXT_PUBLIC_API_ADMIN_HOST=  
+	#endpoint, e.g., query that will be invoked when running SPARQL query
+	NEXT_PUBLIC_API_QUERY_ENDPOINT= 
+	````
+
+	**<span style="color: red;">Note:</span>** For Google OAuth, unless the ```Publishing status``` is ```published```, it will be ```Testing``` as shown in {numref}`google_oauth_setup`. Since this is in a testing mode, only the testing users can log in via Google OAuth. 
+
+	If the ```User type``` is ```Internal```, only the organizational members can log in (e.g., *.mit.edu).
+
+	```{figure} oauth_setup_google.png
+	:name: google_oauth_setup
+	Google OAuth setup.
+	```
+- The next step is to configure the `config-home.yaml` and `config-knowledgebases.yaml` inside `components` directory. The `config-home.yaml` is use to configure the home (or landing) page and `config-knowledgebases.yaml` is use to configure the `knowledge base` page. {numref}`brainkbdocs-headerconfig` and {numref}`brainkbdocs-statistics_structured_box` show the configuration of the BrainKB UI landing page header using the `config-home.yaml` file. Similarly, {numref}`brainkbdocs-knowledgebasepage` shows the configuration of the `knowledge base` page using `config-knowledgebases.yaml` and the mappings of the configuration files into different UI compoentns of `knowledge base` page.
```

Comment:
i don't have `components` directory. It would be also good to have the templates of the files somewhere that could be downloaded

Reply:
`components` should be inside `src/app`. Could you check and make sure that it isn't there?

<img width="1492" alt="image" src="https://github.com/user-attachments/assets/05772720-65eb-4e25-9af0-dd146a95af20">

Reply:
Also, the default config files including the one that I use for test are already present in the repo.
<img width="1492" alt="Screenshot 2024-08-02 at 11 19 09 AM" src="https://github.com/user-attachments/assets/6df9b468-d36f-410e-b1ba-0122177cc032">